perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2857,4 +2838,37 @@ private boolean checkGroupsCycle(PerunSession sess, int resultGroupId, int opera
 
 		return false;
 	}
+
+	/**
+	 * Method sets subGroups names by their parent group
+	 * Private method for moving groups
+	 *
+	 * @param sess perun session
+	 * @param subGroups groups with same parent group
+	 * @throws InternalErrorException
+	 */
+	private void setSubGroupsNames(PerunSession sess, List<Group> subGroups) throws InternalErrorException {
```

Comment:
I think this method can be processed recursively. On every subgroup you can call this method for all it's subgroups until any subgroups exists. What do you think? Or you can use method 'getAllSubGroups' and then order them by name from the shortest to the longest and for every such group change the name, because you can be sure, that parent group has always shorter name then it's subgroup.

Reply:
I believe it already uses recursion. We probably could process all sub group at once, but I'm not sure now `just sorting` would work (maybe yes, since we limit possible characters for group names).
But goal here is to modify part of name for all subgroups, shared prefix actually. If you would get old and new prefix on method input, you could perform string replace on all subgroups name prefixes at once (without recursion) and then store them to DB.
But current implementation is clear/readable, which is also benefit.
I would just optimize this and pass also "parentGroup" to the method input, so you don't have to call ``"getParentGroup(sess, subGroup)`` for each subgroup you are renaming. Because you already fixed the name of parent group in previous method run.